lib/shared/bucketing-assembly-script/assembly/eventQueue/requestPayloadManager.ts

```diff
@@ -28,17 +33,19 @@ export class RequestPayloadManager {
         aggEventQueue: AggEventQueue
     ): FlushPayload[] {
         this.checkForFailedPayloads()
+        const records = new Array<UserEventsBatchRecord>()
 
         // Add events from user event queue
         const userEventQueueKeys = userEventQueue.keys()
         for (let i = 0; i < userEventQueueKeys.length; i++) {
             const key = userEventQueueKeys[i]
             const userEventsRecord = userEventQueue.get(key)
-            this.addEventsToPendingPayloads(userEventsRecord)
+            records.push(userEventsRecord)
         }
 
         // Add events from aggregate events
-        this.addAggEventsToPendingPayloads(aggEventQueue)
+        records.push(this.addAggEventsToPendingPayloads(aggEventQueue))
```

Comment:
you should probably rename this method now that its purpose has changed

`constructBatchRecordsFromAggEvents` or something